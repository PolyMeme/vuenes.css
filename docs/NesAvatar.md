# nes-avatar 

Vue Component implementation of the Nes Avatar 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## props 

- `src` ***String*** (*required*) 

  Define the src link of the image just like `<img>` 

- `alt` ***String*** (*optional*) `default: Empty String` 

  Defines the alt text of the `<img>` 

- `small` ***Boolean*** (*optional*) `default: false` 

  Define if the Avatar is small. Will use the `is-small` class 

- `medium` ***Boolean*** (*optional*) `default: false` 

  Define if the Avatar is medium. Will use the `is-medium` class 

- `large` ***Boolean*** (*optional*) `default: false` 

  Define if the Avatar is large. Will use the `is-large` class 

- `rounded` ***Boolean*** (*optional*) `default: false` 

  Define if the Avatar is rounded. Will use the `is-rounded` class 

## computed properties 

- `nesAvatarClasses` 

  Calculate CSS Classes to use depending on which prop you use 

   **dependencies:** `small`, `medium`, `large`, `rounded` 


