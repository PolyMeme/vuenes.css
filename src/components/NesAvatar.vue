<template>
  <img
    :src="src"
    :alt="alt"
    class="nes-avatar"
    :class="nesAvatarClasses"
  >
</template>

<script>
export default {
  props: {
    src: {
      type: String,
      required: true
    },
    alt: {
      type: String,
      default: () => ''
    },
    small: Boolean,
    medium: Boolean,
    large: Boolean,
    rounded: Boolean
  },
  computed: {
    nesAvatarClasses () {
      return {
        'is-small': this.small,
        'is-medium': this.medium,
        'is-large': this.large,
        'is-rounded': this.rounded
      }
    }
  }
}
</script>
