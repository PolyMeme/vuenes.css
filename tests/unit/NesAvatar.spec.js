import { mount } from '@vue/test-utils'
import NesAvatar from '../../src/components/NesAvatar.vue'

describe('NesAvatar.vue', () => {
  const createAvatar = (propsData) => {
    return mount(NesAvatar, {
      propsData: {
        src: 'http://lorempixel.com/g/500/500',
        ...propsData
      }
    })
  }

  it('should render avatar properly', () => {
    const wrapper = createAvatar()

    expect(wrapper.find('img.nes-avatar').exists()).toBeTruthy()
  })

  it('should render avatar with alt', () => {
    const wrapper = createAvatar({
      alt: 'This is a unit Test!'
    })

    expect(wrapper.find('img.nes-avatar').attributes().alt).toBe('This is a unit Test!')
  })

  it('should add is-rounded class if rounded prop is true', () => {
    const wrapper = createAvatar({
      rounded: true
    })

    expect(wrapper.find('img.nes-avatar.is-rounded').exists()).toBeTruthy()
  })

  it('should add is-small class if small prop is true', () => {
    const wrapper = createAvatar({
      small: true
    })

    expect(wrapper.find('img.nes-avatar.is-small').exists()).toBeTruthy()
  })

  it('should add is-medium class if medium prop is true', () => {
    const wrapper = createAvatar({
      medium: true
    })

    expect(wrapper.find('img.nes-avatar.is-medium').exists()).toBeTruthy()
  })

  it('should add is-large class if large prop is true', () => {
    const wrapper = createAvatar({
      large: true
    })

    expect(wrapper.find('img.nes-avatar.is-large').exists()).toBeTruthy()
  })

  it('should match snapshot', () => {
    const wrapper = createAvatar({
      alt: 'this is a unit test!',
      large: true,
      rounded: true
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
