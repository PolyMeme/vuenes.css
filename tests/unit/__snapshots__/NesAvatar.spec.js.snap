// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesAvatar.vue should match snapshot 1`] = `<img src="http://lorempixel.com/g/500/500" alt="this is a unit test!" class="nes-avatar is-large is-rounded">`;
